source/upgrade.txt

```diff
@@ -0,0 +1,493 @@
+  section.
+- You must add an explicit dependency on the ``org.bson.codecs.record``
+  module if your application deploys the driver in an OSGi container and
+  relies on the driver for encoding and decoding Java records.
+
+- The ``RecordCodec``, implemented in v4.6, deserialized POJOs and record
+  classes that are specified as type parameters of ``List`` or ``Map`` fields
+  of a record as ``Document`` values instead of their respective classes. This
+  version now deserializes them to the proper record and POJO types.
+
+  For example, the following record class definitions show a ``Book`` record
+  that contains a ``List`` that receives a ``Chapter`` type parameter:
+
+  .. code-block:: java
+
+     public record Book(String title, List<Chapter> chapters) {}
+     public record Chapter(Integer number, String text) {}
+
+  Starting in this version, the codec deserializes data in the ``List`` into
+  ``Chapter`` record classes instead of ``Document`` values.
+
+.. _scala-breaking-changes-v4.7:
+
+Version 4.7 Breaking Changes
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This driver version introduces the following breaking changes:
+
+- The ``setWindowFields`` builder API is no longer beta. The new builder
+  breaks binary and source compatibility. See the
+  `Aggregates API documentation <https://mongodb.github.io/mongo-java-driver/4.7/apidocs/mongodb-driver-core/com/mongodb/client/model/Aggregates.html>`__
+  for information the new ``setWindowFields()`` method signatures.
+
+  If your application uses this builder in a version earlier than v4.7, update
+  your source code to use the new method signature and rebuild your binary.
+
+.. _scala-breaking-changes-v4.2:
+
+Version 4.2 Breaking Changes
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This driver version introduces the following breaking changes:
+
+- The ``ObjectId`` class and its ``serialVersionUID`` field were updated
+  to use a new format that minimizes serialization compatibility issues
+  across different versions of the driver.
+
+  If an application using this version or later of the driver attempts to
+  perform Java Object Serialization on any objects that contain an
+  ``ObjectId`` and were serialized by a prior version of the driver, Java
+  throws an ``InvalidClassException``.
+
+  To learn more about Java Object Serialization, see the Java
+  Documentation on `Serializable Objects <https://docs.oracle.com/javase/tutorial/jndi/objects/serial.html>`__.
+
+.. _scala-breaking-changes-v4.0:
+
+Version 4.0 Breaking Changes
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This driver version introduces the following breaking changes:
+
+- Several classes and methods marked as deprecated in the 3.12 release
+  were removed in this version.
+- The insert helper methods return an insert result object instead of
+  ``void``.
+- The ``toJson()`` methods on ``BsonDocument``, ``Document``, and
+  ``DbObject`` return a relaxed JSON format instead of a strict JSON
```

Comment:
```suggestion
- The ``toJson()`` methods in the ``BsonDocument``, ``Document``, and
  ``DbObject`` classes return a relaxed JSON format instead of a strict JSON
```